src/main/java/org/jabref/logic/shared/DBMSConnectionProperties.java

```diff
@@ -117,7 +117,13 @@ public void setUseSSL(boolean useSSL) {
     }
 
     public String getUrl() {
-        return type.getUrl(host, port, database);
+        String url = type.getUrl(host, port, database);
+        if (this.type == DBMSType.MYSQL) {
+            // quickfix - see https://mysqlconnector.net/troubleshooting/retrieval-public-key/
+            url += "?allowPublicKeyRetrieval=true";
+            url += "&useSSL=" + this.isUseSSL();
```

Comment:
Please extract this allowPublicKey to a property in `DBMSConnectionProperties` and do not use the url for this. All server options are passed as property to the connection manager in `DBMSConnection`.  And SSL is duplicated as it is already there.

`            this.connection = DriverManager.getConnection(connectionProperties.getUrl(), connectionProperties.asProperties());
`